builtin/logical/pki/path_fetch_issuers.go

```diff
@@ -0,0 +1,252 @@
+their identifier and their name (if set).
+`
+)
+
+func pathGetIssuer(b *backend) *framework.Path {
+	pattern := "issuer/" + framework.GenericNameRegex("ref") + "(/der|/pem)?"
+	return buildPathGetIssuer(b, pattern)
+}
+
+func buildPathGetIssuer(b *backend, pattern string) *framework.Path {
+	return &framework.Path{
+		// Returns a JSON entry.
+		Pattern: pattern,
+		Fields: map[string]*framework.FieldSchema{
+			"ref": {
+				Type:        framework.TypeString,
+				Description: `Reference to issuer; either "default" for the configured default issuer, an identifier of an issuer, or the name assigned to the issuer.`,
+				Default:     "default",
+			},
+			"name": {
+				Type:        framework.TypeString,
+				Description: `Human-readable name for this issuer.`,
+			},
+		},
+
+		Callbacks: map[logical.Operation]framework.OperationFunc{
+			logical.ReadOperation:   b.pathGetIssuer,
+			logical.UpdateOperation: b.pathUpdateIssuer,
+			logical.DeleteOperation: b.pathDeleteIssuer,
+		},
+
+		HelpSynopsis:    pathGetIssuerHelpSyn,
+		HelpDescription: pathGetIssuerHelpDesc,
+	}
+}
+
+func (b *backend) pathGetIssuer(ctx context.Context, req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	// Handle raw issuers first.
+	if strings.HasSuffix(req.Path, "/der") || strings.HasSuffix(req.Path, "/pem") {
+		return b.pathGetRawIssuer(ctx, req, data)
+	}
+
+	issuerName := data.Get("ref").(string)
+	if len(issuerName) == 0 {
+		return logical.ErrorResponse("missing issuer reference"), nil
+	}
+
+	ref, err := resolveIssuerReference(ctx, req.Storage, issuerName)
+	if err != nil {
+		return nil, err
+	}
+	if ref == "" {
+		return logical.ErrorResponse("unable to resolve issuer id for reference: " + issuerName), nil
+	}
+
+	issuer, err := fetchIssuerById(ctx, req.Storage, ref)
+	if err != nil {
+		return nil, err
+	}
+
+	return &logical.Response{
+		Data: map[string]interface{}{
+			"id":          issuer.ID,
+			"name":        issuer.Name,
+			"key_id":      issuer.KeyID,
+			"certificate": issuer.Certificate,
+		},
+	}, nil
+}
+
+func (b *backend) pathUpdateIssuer(ctx context.Context, req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	issuerName := data.Get("ref").(string)
+	if len(issuerName) == 0 {
+		return logical.ErrorResponse("missing issuer reference"), nil
+	}
+
+	newName := data.Get("name").(string)
+	if len(newName) > 0 && !nameMatcher.MatchString(newName) {
+		return logical.ErrorResponse("new issuer name outside of valid character limits"), nil
+	}
+
+	ref, err := resolveIssuerReference(ctx, req.Storage, issuerName)
+	if err != nil {
+		return nil, err
+	}
+	if ref == "" {
+		return logical.ErrorResponse("unable to resolve issuer id for reference: " + issuerName), nil
+	}
+
+	issuer, err := fetchIssuerById(ctx, req.Storage, ref)
+	if err != nil {
+		return nil, err
+	}
+
+	if newName != issuer.Name {
+		issuer.Name = newName
+
+		err := writeIssuer(ctx, req.Storage, issuer)
```

Comment:
I have always been torn about this, but since we are passing around pointers to structs, shouldn't we write out a new Issuer object instead of updating an existing one?

Reply:
Not sure I follow why? I've actually considered restructuring to something like `issuer.Write()`.

If you write out a new object, you've gotta copy _everything_ over, and if the fields change, that seems like a brittle problem waiting to happen. Updating the value seems easiest? Why would pointer versus not matter?

Reply:
Keep it as is for now, like I said I'm torn about this, same thing as before about returning pointers to structs vs structs themselves. You raise a descent point about field addition and my counters to that at this time don't really apply to this codebase at the moment.

I'd rather not have the issuer.Write() though that is for sure, just have a method that takes an issuer to write out like we have now.